### ISSUE TITLE
FATAL ERROR: Ineffective mark-compacts near heap limit Allocation failed - JavaScript heap out of memory

### ISSUE THREAD

**Issue by AlexZalyalowa** [kubevious/kubevious#49]
Hello.
I have an empty Universe in ui and parser sometimes crushes with the error
```
FATAL ERROR: Ineffective mark-compacts near heap limit Allocation failed - JavaScript heap out of memory

<--- Last few GCs --->

[1:0x561c150a5f00]   224796 ms: Mark-sweep 1761.5 (2068.0) -> 1746.9 (2068.5) MB, 2813.1 / 0.8 ms  (average mu = 0.090, current mu = 0.023) allocation failure scavenge might not succeed
[1:0x561c150a5f00]   227659 ms: Mark-sweep 1762.1 (2068.5) -> 1747.4 (2069.2) MB, 2814.7 / 0.6 ms  (average mu = 0.054, current mu = 0.017) allocation failure scavenge might not succeed


<--- JS stacktrace --->

==== JS stack trace =========================================

    0: ExitFrame [pc: 0x561c11d5d999]
Security context: 0x1d7b2f5808d1 <JSObject>
    1: sort [0x1d7b2f59a889](this=0x1fbd54989a29 <JSArray[12]>,0x3ed22bf404b1 <undefined>)
    2: stringify(aka stringify) [0x39fb9f396c41] [/app/node_modules/fast-json-stable-stringify/index.js:~19] [pc=0x2b17cb8d8df5](this=0x3ed22bf404b1 <undefined>,0x39fb9f3961e1 <Object map = 0x6e5bb02bf49>)
    3: stringify(aka stringify) [0x39fb9f396c41] [/app/node_mod...
```
May be it's because my cluster is pretty big (27 nodes). I've tried to increase parser and kubevious replicas up to 7, but it didn't help.

**Comment by rubenhak** [comment]
@AlexZalyalowa, scaling horizontally (number of pods) wouldn't help. Lets scale them down to 1 of each. Instead, they should be scaled vertically. First of all increase memory allocation to parser and kubevious significantly - 2GB is not sufficient.

**Comment by AlexZalyalowa** [comment]
@rubenhak 
I've increased parser's memory up to 20Gb, kubevious's memory up to 10Gb and now universe is still empty and parser has this error in logs
```
/app/lib/concrete/registry.js:62
            delete this._itemsKindDict[item.groupKey][rawId];
                                                      ^

TypeError: Cannot convert undefined or null to object
    at ConcreteRegistry.remove (/app/lib/concrete/registry.js:62:55)
    at K8sParser.parse (/app/lib/parsers/k8s.js:24:44)
    at K8sLoader._handle (/app/lib/loaders/k8s.js:169:33)
    at ResourceWatch._cb (/app/lib/loaders/k8s.js:99:18)
    at ResourceWatch._handleChange (/app/node_modules/k8s-super-client/lib/resource-watch.js:148:18)
    at DestroyableTransform.<anonymous> (/app/node_modules/k8s-super-client/lib/resource-watch.js:105:30)
    at DestroyableTransform.emit (events.js:315:20)
    at addChunk (/app/node_modules/readable-stream/lib/_stream_readable.js:291:12)
    at readableAddChunk (/app/node_modules/readable-stream/lib/_stream_readable.js:278:11)
    at DestroyableTransform.Readable.push (/app/node_modules/readable-stream/lib/_stream_readable.js:245:10)
    at DestroyableTransform.Transform.push (/app/node_modules/readable-stream/lib/_stream_transform.js:148:32)
    at push (/app/node_modules/split2/index.js:52:10)
    at DestroyableTransform.transform [as _transform] (/app/node_modules/split2/index.js:33:5)
    at DestroyableTransform.Transform._read (/app/node_modules/readable-stream/lib/_stream_transform.js:184:10)
    at DestroyableTransform.Transform._write (/app/node_modules/readable-stream/lib/_stream_transform.js:172:83)
    at doWrite (/app/node_modules/readable-stream/lib/_stream_writable.js:428:64)
```

**Comment by rubenhak** [comment]
@AlexZalyalowa, we have to take a look at this deeper. It would be easier if you join the slack channel to discuss.

**Comment by kubevious** [comment]
We have a potential fix in [v0.8](https://github.com/kubevious/kubevious/releases/tag/v0.8). Please try it out.

**Comment by kubevious** [comment]
@AlexZalyalowa, more improvements were made to support production scale setups. 

For your large setups try increasing the following variables in the chart. Try giving at least 4GB to each. 
- collector.resources.requests.*
- collector.resources. limits.*
- parser.resources.requests.*
- parser.resources. limits.*

Please try installing again using the latest chart and let us know how it works. Reopen this issue if it still persists.
```sh
kubectl create namespace kubevious

helm repo add kubevious https://helm.kubevious.io

helm repo update

helm upgrade --atomic -i kubevious kubevious/kubevious --version 0.9.27 -n kubevious
```
For more details on installation options, visit [Deployment Repository](https://github.com/kubevious/helm).
